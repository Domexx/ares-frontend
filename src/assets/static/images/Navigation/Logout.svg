<svg width="21" height="21" viewBox="0 0 21 21" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_1576_12)">
<path d="M4.50348 19.5193H7.00613C7.69827 19.5193 8.25745 18.9601 8.25745 18.2679C8.25745 17.5758 7.69827 17.0166 7.00613 17.0166H4.50348C3.81252 17.0166 3.25216 16.4563 3.25216 15.7653V5.7547C3.25216 5.06374 3.81252 4.50338 4.50348 4.50338H7.00613C7.69827 4.50338 8.25745 3.94302 8.25745 3.25206C8.25745 2.56109 7.69827 2.00073 7.00613 2.00073H4.50348C2.4302 2.00073 0.749512 3.68142 0.749512 5.7547V15.7653C0.749512 17.8378 2.4302 19.5193 4.50348 19.5193ZM20.4891 10.0757L14.8543 4.75756C14.5815 4.50033 14.1817 4.43006 13.8376 4.57791C13.4933 4.72639 13.2702 5.06554 13.2702 5.44015L13.2666 8.25563L8.26129 8.25559C7.56955 8.25559 7.00919 8.81595 7.00919 9.50692V12.0096C7.00919 12.7005 7.56994 13.2609 8.26129 13.2609L13.2666 13.2609L13.2702 16.0412C13.2702 16.4157 13.4933 16.7549 13.8376 16.9035C14.1818 17.0513 14.5817 16.981 14.8543 16.7238L20.4891 11.4057C20.8645 11.0885 20.8645 10.4315 20.4891 10.0757Z" fill="black"/>
</g>
<defs>
<clipPath id="clip0_1576_12">
<rect width="20.0212" height="20.0212" fill="white" transform="translate(0.749512 0.74939)"/>
</clipPath>
</defs>
</svg>
