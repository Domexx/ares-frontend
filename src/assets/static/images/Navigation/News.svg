<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16 0H4C1.791 0 0 1.791 0 4V12C0 14.209 1.791 16 4 16H16C18.209 16 20 14.209 20 12V4C20 1.791 18.209 0 16 0ZM18 12C18 13.103 17.103 14 16 14H4C2.897 14 2 13.103 2 12V4C2 2.897 2.897 2 4 2H16C17.103 2 18 2.897 18 4V12Z" fill="#0E1317"/>
<path d="M0 10V12H20V10H0ZM5 20H15V18H5V20Z" fill="#0E1317"/>
</svg>
