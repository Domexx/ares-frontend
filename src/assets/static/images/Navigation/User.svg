<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M10 10C13.0689 10 15.5556 7.762 15.5556 5C15.5556 2.238 13.0689 0 10 0C6.93111 0 4.44444 2.238 4.44444 5C4.44444 7.761 6.93111 10 10 10ZM10 2C11.8378 2 13.3333 3.346 13.3333 5C13.3333 6.654 11.8378 8 10 8C8.16222 8 6.66667 6.654 6.66667 5C6.66667 3.346 8.16222 2 10 2ZM13.3333 12H6.66667C2.98556 12 0 14.686 0 18V20H2.22222V18C2.22222 15.795 4.21556 14 6.66667 14H13.3333C15.7844 14 17.7778 15.795 17.7778 18V20H20V18C20 14.686 17.0144 12 13.3333 12Z" fill="#B5B6B8"/>
</svg>
