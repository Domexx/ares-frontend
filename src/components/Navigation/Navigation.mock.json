{
  "main": [
    {
      "id": "cosmic_news",
      "name": "News",
      "url": "news",
      "icon": "news",
      "children": []
    },
    {
      "id": "cosmic_statistics",
      "name": "Statistics",
      "url": "statistics",
      "icon": "statistics",
      "children": []
    },
    {
      "id": "cosmic_users",
      "name": "Users",
      "url": "users",
      "icon": "users",
      "children": []
    }
  ]
}
