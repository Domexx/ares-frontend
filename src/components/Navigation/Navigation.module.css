/* === === === === Navigation === === === === */

.Navigation {
  position: relative;
  flex: 0 0 10%;
  order: -1;
  width: 100%;
  min-width: 120px;
  max-width: 120px;
  min-height: 100%;
  margin: 0;
  padding: 40px 16px;
  background: var(--clr-dark-400);
  transition: all 250ms ease-in-out;
}

.Navigation > ul > li > a {
  margin-left: 30px;
  white-space: nowrap;
}

.Navigation.is-Active {
  min-width: 300px;
  max-width: 300px;
}

.Navigation:not(.is-Active) > ul > li > a {
  margin-left: 0;
}

.ListItem:not(.is-Active) {
  flex-direction: column;
  justify-content: center;
  transition: all 250ms ease-in-out;
}

.ListItem:not(.is-Active) + .ListItem:not(.is-Active) {
  margin-top: 20px;
}

.Avatar:not(.is-Active) > span,
.Avatar:not(.is-Active) img {
  min-width: 90px;
  transition: all 250ms ease-in-out;
}

@media (--laptop) {
  .Navigation.is-Active {
    min-width: 280px;
    max-width: 280px;
  }

  .Navigation {
    height: auto;
  }
}

@media (--phone) {
  .Navigation {
    position: fixed;
    left: 0;
    transform: translateX(-100%);
  }

  .Navigation.is-Active {
    position: fixed;
    z-index: 100;
    left: 0;
    min-width: 100%;
    height: 100%;
    transform: translateX(0);
  }
}
