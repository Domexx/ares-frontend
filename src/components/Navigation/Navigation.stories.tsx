import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import NavigationComponent, { Props } from './Navigation';

import MenuMock from './Navigation.mock.json';

const Component = {
  title: 'Components/Navigation',
  component: NavigationComponent,
};

const Template: Story<ComponentProps<typeof NavigationComponent>> = (args) => {
  return <NavigationComponent {...args} />;
};

export const Navigation = Template.bind({});
export default Component;

const { main } = MenuMock;

// https://storybook.js.org/docs/react/essentials/controls
Navigation.args = {
  main,
} as Props;
