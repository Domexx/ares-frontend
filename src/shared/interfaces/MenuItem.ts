// Generated by https://quicktype.io

export interface Menu {
  main: MenuItem[];
}

export interface MenuItem {
  id: string;
  name: string;
  url: string;
  icon: string;
  locale: boolean;
  children?: MenuItem[];
}
