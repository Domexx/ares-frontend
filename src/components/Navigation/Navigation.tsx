import React from 'react';
import s from './Navigation.module.css';

import { MenuItem } from '../../shared/interfaces/MenuItem';
import List from './List';

export type Props = {
  main: MenuItem[];
};

const Navigation: React.FC<Props> = ({ main }) => {
  return (
    <nav className={s['navigation']}>
      <List items={main} />
    </nav>
  );
};

export default Navigation;
