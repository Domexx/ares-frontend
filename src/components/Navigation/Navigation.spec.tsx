import React from 'react';

import { render } from '@testing-library/react';

import Navigation from './Navigation';

import Menu from './Navigation.mock.json';
import { MenuItem } from '../../shared/interfaces/MenuItem';

describe('Navigation Component', () => {
  let mainMock: MenuItem[];

  beforeAll(() => {
    const { main } = Menu;
    mainMock = main;
  });

  it('should render properly', () => {
    const { container } = render(
      <Navigation
        main={mainMock}
        avatar={{
          image: '/test.png',
          name: 'Dome',
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
