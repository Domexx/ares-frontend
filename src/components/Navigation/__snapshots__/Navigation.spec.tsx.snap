// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render properly 1`] = `
<nav
  class="navigation"
/>
`;
